### PR TITLE
Remove polyfills and drop IE11 support

### LINKED ISSUE #59: Failing to display Ad in React environments, also crashing the websites
Observing this issue from a week. I am able to reproduce the issue in the following code sandbox environment. 

https://codesandbox.io/s/hopeful-leaf-b8s5i?file=/src/Ad.js:386-859

To reproduce please comment and uncomment the `React.useEffect` snippet i.e from line # 17 to # 34

```
Possible Unhandled Promise Rejection: 
TypeError: Failed to construct 'URL': Please use the 'new' operator.
    at new s (https://media.ethicalads.io/media/client/ethicalads.min.js:1:17327)

    at https://media.ethicalads.io/media/client/ethicalads.min.js:1:18403

    at new s (https://media.ethicalads.io/media/client/ethicalads.min.js:1:18464)

    at e.value (https://media.ethicalads.io/media/client/ethicalads.min.js:1:11362)

    at e.value (https://media.ethicalads.io/media/client/ethicalads.min.js:1:10473)

    at https://media.ethicalads.io/media/client/ethicalads.min.js:1:13495

    at Array.map (<anonymous>)

    at _ (https://media.ethicalads.io/media/client/ethicalads.min.js:1:13248)

    at https://media.ethicalads.io/media/client/ethicalads.min.js:1:14271

    at https://media.ethicalads.io/media/client/ethicalads.min.js:1:1476

    at https://codesandbox.io/static/js/vendors~react-devtools-backend.d7ac7a663.chunk.js:1:62677

    at c (https://codesandbox.io/static/js/vendors~react-devtools-backend.d7ac7a663.chunk.js:1:62792)

    at n (https://codesandbox.io/static/js/vendors~react-devtools-backend.d7ac7a663.chunk.js:1:61737)
```

Screenshot 2021-04-30 at 12 24 38 PM

### LINKED ISSUE #62: Use ponyfills instead of polyfills to not change state on others' sites

### PR DESCRIPTION
**Pull request by davidfischer**
We had some issues with the polyfills on some sites but I don't think we care about IE11 support enough to keep these.

- Shrinks the minified client from 40k -> 25k (should be ~20% of these gzipped)
- Gracefully fails (without showing an ad) on client that aren't supported. Clients require `URL`, `URLSearchParams`, `Promise`, and `Object.entries`.

Fixes #62
Fixes #59

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -527,6 +525,27 @@ export class Placement {
...
+ *  @returns {boolean} true if all dependencies met and false otherwise
+ */
+export function check_dependencies() {
+  if (
+    !Object.entries ||
+    !window.URL ||
+    !window.URLSearchParams ||
+    !window.Promise
+  ) {
+    console.error("Browser does not meet ad client dependencies.");
```
Comment (ericholscher) [inline index.js:542]:
```suggestion
    console.error("Browser does not meet ethical ad client dependencies. Not showing ads");
```

**Review by ericholscher — APPROVED** [review]
Smaller bundle FTW. Seems simple enough 👍 

As an aside: I'd love to have better error messages for when we don't return an ad. Currently we just say like "ad client not configured correctly", but often it's an issue with the server (eg. no ads in the flight). Might be worth improving that error at some point in the future as well.

**Comment by davidfischer** [comment]
> As an aside: I'd love to have better error messages for when we don't return an ad. Currently we just say like "ad client not configured correctly", but often it's an issue with the server (eg. no ads in the flight). Might be worth improving that error at some point in the future as well.

I filed an issue for this a while back https://github.com/readthedocs/ethical-ad-client/issues/26.
This won't happen when there's no ads in the flight though. Flights aren't eligible to be chosen if there's no ads in them.

**Comment by nienn** [comment]
I have no specific thought on dropping IE11 support for the Ethical Ads, as I have no knowledge of its policies or user base but, overall, I'm up for dropping IE11 when it envolves too many otherwise unnecessary resources and bugfixes.

With that in mind, this seems like a great change!  🚀

**Pull request merged**